idaes/generic_models/properties/core/eos/ceos.py

```diff
@@ -812,6 +812,16 @@ def gibbs_mol_phase_comp(b, p, j):
                 b.entr_mol_phase_comp[p, j] *
                 b.temperature)
 
+    @staticmethod
+    def vol_mol_phase(b, p):
+        pobj = b.params.get_phase(p)
+        if pobj.is_vapor_phase() or pobj.is_liquid_phase():
+            return (Cubic.gas_constant(b)*b.temperature *
+                    b.compress_fact_phase[p] /
+                    b.pressure)
```

Comment:
What is compress_fact_phase?

I'm a little confused how this works for incompressible liquids. I would have thought that it would be calculated just based on the dens_mass and molecular weight. If compress_fact_phase is defined for an incompressible liquid in such a way that it cancels out the gas_constant, temperature, and pressure, and it basically just comes down to dens_mass and molecular weight isn't this just making two relatively complicated constraints and an additional variable?

Reply:
It is the compressibility factor, Z, which is calculated by the equation of state: `PV=ZRT`. The whole purpose of the equation of state is to calculate `Z` for the current state.